controllers/datamovement_controller.go

```diff
@@ -86,314 +83,237 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
+		"--host", strings.Join(hosts, ","),
+		"dcp", dm.Spec.Source.Path, dm.Spec.Destination.Path)
 
-		if err := r.Status().Update(ctx, dm); err != nil {
-			log.Error(err, "Failed to transition to running state")
-			return ctrl.Result{}, err
-		}
+	log.Info("Running Commands", "cmd", cmd.String())
 
-		return ctrl.Result{Requeue: true}, nil
+	// TODO: Capture output as it progresses and add a %complete to the resource
 
-	case nnfv1alpha1.DataMovementConditionTypeRunning:
-		monitorFn := map[bool]func(context.Context, *nnfv1alpha1.NnfDataMovement) (*ctrl.Result, string, string, error){
-			false: r.monitorRsyncJob,
-			true:  r.monitorLustreJob,
-		}[isLustre2Lustre]
+	out, err := cmd.CombinedOutput()
 
-		result, status, message, err := monitorFn(ctx, dm)
-		log.Info("Monitor job", "status", status, "message", message)
+	now := metav1.NowMicro()
+	dm.Status.EndTime = &now
+	dm.Status.State = nnfv1alpha1.DataMovementConditionTypeFinished
+	dm.Status.Status = nnfv1alpha1.DataMovementConditionReasonSuccess
 
-		if err != nil {
-			log.Error(err, "Failed to monitor")
+	if err != nil {
+		dm.Status.Status = nnfv1alpha1.DataMovementConditionReasonFailed
+		dm.Status.Message = err.Error()
 
-			msg := fmt.Sprintf("Failed to monitor: %v", err)
-			resetConditionRollup(dm, nnfv1alpha1.DataMovementConditionReasonFailed, msg)
+		// TODO: Enhanced error capture: parse error response and provide useful message
 
-			if err := r.Status().Update(ctx, dm); err != nil {
-				log.Error(err, "Failed to record monitor error status")
-				return ctrl.Result{}, err
-			}
+		log.Error(err, "Data movement operation failed", "output", string(out))
+	}
 
-			return ctrl.Result{}, err
-		} else if result != nil && !result.IsZero() {
-			return *result, nil
-		}
+	status := dm.Status.DeepCopy()
 
-		// Continue monitoring the resource, if specified, by preventing the data movement resource from finishing
-		// early while the client application is running and generating data movement requests. This is to protect
-		// the use case where this resource thinks it handled all the requests but in reality the client application
-		// continues to execute with the potential to generate more requests.
-		if dm.Spec.Monitor == true {
-			return ctrl.Result{}, nil
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		dm := &nnfv1alpha1.NnfDataMovement{}
+		if err := r.Get(ctx, req.NamespacedName, dm); err != nil {
+			return client.IgnoreNotFound(err)
 		}
 
-		switch status {
-		case nnfv1alpha1.DataMovementConditionTypeRunning:
-			// Still running, nothing to do here
-			break
-		case nnfv1alpha1.DataMovementConditionReasonFailed, nnfv1alpha1.DataMovementConditionReasonSuccess:
+		status.DeepCopyInto(&dm.Status)
 
-			advanceCondition(dm,
-				nnfv1alpha1.DataMovementConditionTypeFinished,
-				status,
-				message,
-			)
+		return r.Status().Update(ctx, dm)
+	})
 
-			now := metav1.NowMicro()
-			dm.Status.EndTime = &now
-		}
+	if errors.IsConflict(err) {
```

Comment:
We still need to catch other errors: https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict

>  May be conflict if max retries were hit, or may be something unrelated like permissions or a network error

Reply:
yes, I forgot to return `err`